File: domo_backend/app/models/activity.py
from datetime import datetime
from typing import Optional
from sqlmodel import SQLModel, Field, Relationship


class ActivityLog(SQLModel, table=True):
    __tablename__ = "activity_logs"

    id: Optional[int] = Field(default=None, primary_key=True)

    user_id: int = Field(foreign_key="users.id", index=True)      # 누가
    workspace_id: Optional[int] = Field(default=None, foreign_key="workspaces.id", index=True) # 어디서

    content: str  # 내용 (예: "김철수님이 새 프로젝트 '알파'를 생성했습니다.")
    action_type: str # 분류 (CREATE, UPDATE, DELETE, UPLOAD 등) - 필터링용

    created_at: datetime = Field(default_factory=datetime.now)
    workspace: Optional["Workspace"] = Relationship(back_populates="activity_logs")

File: domo_frontend/src/models/api/activity.ts
import { API_CONFIG, apiFetch, mockDelay } from './config';

// ============================================
// 활동 로그 타입
// ============================================

export interface ActivityLog {
  id: number;
  user_id: number;
  content: string;
  action_type: string;
  created_at: string;
}

// ============================================
// 활동 로그 API
// ============================================

/**
 * 내 활동 로그 조회
 */
export async function getMyActivities(): Promise<ActivityLog[]> {
  if (API_CONFIG.USE_MOCK) {
    await mockDelay(200);
    return [
      {
        id: 1,
        user_id: 1,
        content: '새로운 카드를 생성했습니다.',
        action_type: 'CREATE',
        created_at: new Date().toISOString(),
      },
      {
        id: 2,
        user_id: 1,
        content: '파일을 업로드했습니다.',
        action_type: 'UPLOAD',
        created_at: new Date(Date.now() - 3600000).toISOString(),
      },
    ];
  }

  // 백엔드: GET /api/users/me/activities
  return apiFetch<ActivityLog[]>('/users/me/activities');
}

/**
 * 워크스페이스 활동 로그 조회
 */
export async function getWorkspaceActivities(workspaceId: number): Promise<ActivityLog[]> {
  if (API_CONFIG.USE_MOCK) {
    await mockDelay(200);
    return [
      {
        id: 1,
        user_id: 1,
        content: '새로운 멤버가 참여했습니다.',
        action_type: 'JOIN',
        created_at: new Date().toISOString(),
      },
    ];
  }

  // 백엔드: GET /api/workspaces/{id}/activities
  return apiFetch<ActivityLog[]>(`/workspaces/${workspaceId}/activities`);
}
